pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -149,116 +202,50 @@ func (r *CloudEntitySelectorReconciler) processCreateOrUpdate(selector *cloudv1a
+	accountNamespacedName *types.NamespacedName) {
 	r.mutex.Lock()
 	defer r.mutex.Unlock()
-
-	selectorNamespacedName := &types.NamespacedName{
-		Namespace: selector.Namespace,
-		Name:      selector.Name,
-	}
-
-	if pollerScope, exists := r.accPollers[*selectorNamespacedName]; exists {
-		r.Log.Info("poller exists", "selector", selectorNamespacedName)
-		return pollerScope, exists
-	}
-
-	accountNamespacedName := &types.NamespacedName{
-		Namespace: selector.Namespace,
-		Name:      selector.Spec.AccountName,
-	}
-	account := &cloudv1alpha1.CloudProviderAccount{}
-	_ = r.Get(context.TODO(), *accountNamespacedName, account)
-	accountCloudType, err := utils.GetAccountProviderType(account)
-	if err != nil {
-		return nil, false
-	}
-	poller := &accountPoller{
-		Client:            r.Client,
-		scheme:            r.Scheme,
-		log:               r.Log,
-		pollIntvInSeconds: *account.Spec.PollIntervalInSeconds,
-		cloudType:         accountCloudType,
-		namespacedName:    accountNamespacedName,
-		selector:          selector.DeepCopy(),
-		ch:                make(chan struct{}),
-	}
-	poller.vmSelector = cache.NewIndexer(
-		func(obj interface{}) (string, error) {
-			m := obj.(*cloudv1alpha1.VirtualMachineSelector)
-			// Create a unique key for each VirtualMachineSelector.
-			return fmt.Sprintf("%v-%v-%v", m.Agented, m.VpcMatch, m.VMMatch), nil
-		},
-		cache.Indexers{
-			virtualMachineSelectorMatchIndexerByID: func(obj interface{}) ([]string, error) {
-				m := obj.(*cloudv1alpha1.VirtualMachineSelector)
-				if len(m.VMMatch) == 0 {
-					return nil, nil
-				}
-				var match []string
-				for _, vmMatch := range m.VMMatch {
-					if len(vmMatch.MatchID) > 0 {
-						match = append(match, strings.ToLower(vmMatch.MatchID))
-					}
-				}
-				return match, nil
-			},
-			virtualMachineSelectorMatchIndexerByName: func(obj interface{}) ([]string, error) {
-				m := obj.(*cloudv1alpha1.VirtualMachineSelector)
-				if len(m.VMMatch) == 0 {
-					return nil, nil
-				}
-				var match []string
-				for _, vmMatch := range m.VMMatch {
-					if len(vmMatch.MatchName) > 0 {
-						match = append(match, strings.ToLower(vmMatch.MatchName))
-					}
-				}
-				return match, nil
-			},
-			virtualMachineSelectorMatchIndexerByVPC: func(obj interface{}) ([]string, error) {
-				m := obj.(*cloudv1alpha1.VirtualMachineSelector)
-				if m.VpcMatch != nil && len(m.VpcMatch.MatchID) > 0 {
-					return []string{strings.ToLower(m.VpcMatch.MatchID)}, nil
-				}
-				return nil, nil
-			},
-		})
-
-	r.accPollers[*selectorNamespacedName] = poller
-
-	r.Log.Info("poller will be created", "selector", selectorNamespacedName)
-	return poller, false
+	r.selectorToAccountMap[*selectorNamespacedName] = *accountNamespacedName
 }
 
-func (r *CloudEntitySelectorReconciler) removeAccountPoller(selectorNamespacedName *types.NamespacedName) *accountPoller {
+func (r *CloudEntitySelectorReconciler) deleteSelectorAccountMapping(selectorNamespacedName *types.NamespacedName) {
```

Comment:
I could not come with up an alternative name, But I think, we can have better name for it.